catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -293,28 +298,76 @@ void registryInfoQuery() throws UnsupportedQueryException {
         SourceResponse identityMetacard = query(queryRequest);
         if (identityMetacard.getResults()
                 .size() > 0) {
-            remoteName = identityMetacard.getResults()
+            String metacardTitle = identityMetacard.getResults()
                     .get(0)
                     .getMetacard()
                     .getTitle();
+            setId(identityMetacard.getResults()
+                    .get(0)
+                    .getMetacard()
+                    .getTitle());
             registryId = RegistryUtility.getRegistryId(identityMetacard.getResults()
                     .get(0)
                     .getMetacard());
+            updateConfiguration(metacardTitle);
         }
-        updateConfiguration();
+
     }
 
-    private void updateConfiguration() {
+    private void updateConfiguration(String metacardTitle) {
+        if (metacardTitle == null) {
+            LOGGER.error("Unable to update registry configurations. No metacard title.");
+            return;
+        }
         String currentPid = getConfigurationPid();
         try {
             Configuration currentConfig = configAdmin.getConfiguration(currentPid);
             Dictionary<String, Object> currentProperties = currentConfig.getProperties();
-            currentProperties.put(REMOTE_NAME, remoteName);
-            currentProperties.put(RegistryObjectMetacardType.REGISTRY_ID, registryId);
-            currentConfig.update(currentProperties);
+            URI uri;
+            String registryUrl = "";
+            if (currentProperties.get("registryUrl") != null) {
+                registryUrl = (String) currentProperties.get("registryUrl");
+            } else if (currentProperties.get("url") != null) {
+                registryUrl = (String) currentProperties.get("url");
+            }else{
+                LOGGER.error("Unable to update registry configurations. No url found");
+            }
+            if (registryUrl != null) {
+                try {
+                    uri = new URI(registryUrl);
+                } catch (URISyntaxException e) {
+                    LOGGER.error(
+                            "Unable to update registry configurations. Syntax error on registryUrl",
+                            e);
+                    return;
+                }
+                StringBuilder nameToSet = new StringBuilder(metacardTitle + " (");
+                StringBuilder hostnamePort = new StringBuilder(uri.getHost());
+                if (uri.getPort() != -1) {
+                    hostnamePort.append(":" + uri.getPort());
+                }
+                String oldName = (String) currentProperties.get("id");
```

Comment:
I know we talked about doing something like this to get the type name but now that I am looking at it I think it would be better just to just create a local constant variable and use it in this construction. It will greatly simplify this bit of code.